### PR TITLE
d3dmetal: Only run sysctl on Sonoma and later

### PR DESCRIPTION
**Pull request by Gcenx**
Closes https://trac.macports.org/ticket/69983

###### Type(s)

- [x] bugfix

###### Tested on

macOS 14.5 23F79 x86_64
Xcode 15.3 15E204a

###### Verification 

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: devel/d3dmetal/Portfile [hunk 1]
```diff
@@ -69,11 +69,13 @@ notes "
 
 platform darwin i386 {
     try {
```
Comment (ryandesign) [inline devel/d3dmetal/Portfile:71]: What is `try` doing? We don't use `try` anywhere else in MacPorts. We do use `catch` a lot…

Comment (jmroot) [reply devel/d3dmetal/Portfile]: `try` by itself (i.e. with no handlers) does nothing, which is why the user in the ticket was seeing the error. `catch { blah }` is more or less equivalent to:
```Tcl
try {
    blah
} on error {} {
}
```

File: devel/d3dmetal/Portfile [hunk 2]
```diff
@@ -68,12 +68,14 @@ notes "
...
-            # Change arch to arm64 to make it unsupported on Intel
-            # yes this is a nasty hack.
-            supported_archs arm64
+    catch {
+        if {${os.major} >= 23} {
+            set is_rosetta2 [exec sysctl -in sysctl.proc_translated]
+            if { ${is_rosetta2} != 1 } {
+                # Change arch to arm64 to make it unsupported on Intel
+                # yes this is a nasty hack.
+                supported_archs arm64
```
[the remark this answers is not in the document]
Comment (jmroot) [reply devel/d3dmetal/Portfile]: `known_fail` is used from the PortIndex, so it won't work correctly to set it based on anything that may differ between the machine generating the index and the one using it.

What would work is something like:
```
if {${os.major} >= 23} {
    pre-fetch {
        set is_rosetta2 [exec sysctl -in sysctl.proc_translated]
        if { ${is_rosetta2} != 1 } {
            error "$name does not work on Intel hardware"
        }
    }
}
```
That has the additional benefit of not running an external program every time the Portfile is parsed.

**Comment by kencu** [comment]
OK, this looks reasonable. Let's kick this can another block down the road.

**Pull request merged**